### PR TITLE
Do math exercise day2

### PR DESCRIPTION
**Pull request by gavin-go-dang**
Completed 5 exercises:
-HackerRank: https://www.hackerrank.com/challenges/the-minion-game/problem?isFullScreen=true
-HackerRank: https://www.hackerrank.com/challenges/find-angle/problem?isFullScreen=true
-Hackeerrank: https://www.hackerrank.com/challenges/merge-the-tools/problem?isFullScreen=true
-HackerRank: https://www.hackerrank.com/challenges/validating-credit-card-number/problem?isFullScreen=true
-HackerRank: https://www.hackerrank.com/challenges/word-order/problem?isFullScreen=true

### CODE REVIEW

File: do_math_exercises/merge_tools.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#Hackeerrank: https://www.hackerrank.com/challenges/merge-the-tools/problem?isFullScreen=true
+
+def merge_the_tools(string, k):
+    def sub_string_to_string(sub_string):
+        sub_string = list(sub_string)
+        set_sub_string = []
+        for char in sub_string:
```
Comment (shinxz12) [inline do_math_exercises/merge_tools.py]: try using comprehension list.

File: do_math_exercises/merge_tools.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+            if char not in set_sub_string:
+                set_sub_string.append(char)
+        return ''.join(set_sub_string) 
+    
+    sub_string = []
+    for i in range(int(len(string) / k)):
```
Comment (shinxz12) [inline do_math_exercises/merge_tools.py:11]: comprehension list

File: do_math_exercises/minion_game.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        if string[i:i+n] not in list_word_kevin and re.match(r'[aeiouAEIOU]', string[i]):
+            list_word_kevin.append(string[i:i+n])
+    return [list_word_kevin, list_word_stuart]
+
+def calculate_point(string):
+    score_kevin = 0
+    score_stuart = 0
+
+    for len_word in range(1, len(string)+1):
+        list_word = get_n_word(string, len_word)
```
Comment (shinxz12) [inline do_math_exercises/minion_game.py]: `a, b = get_n_word(...)`

File: do_math_exercises/valid_creadit_card.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+def check_first_number(digits):
+    if digits[0] not in ['4' ,'5', '6']:
+        return False
+    return True
+
+
+
+def check_len(digits):
+    if len(digits.replace('-', '')) != 16:
```
Comment (shinxz12) [inline do_math_exercises/valid_creadit_card.py]: What's 16 ? This is a magic number. Please comment and define by a constant variable

File: do_math_exercises/valid_creadit_card.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    if not all(list(map(lambda x: True if re.fullmatch(valid_char, x) else False, group_digit))):
+        return False
+    
+    if not all(list(map(lambda x: True if len(x) == 4 else False, group_digit))):
+        return False
+     
+    return True
+
+def check_separator_and_ditgit(credit_number):
+
```
Comment (shinxz12) [inline do_math_exercises/valid_creadit_card.py]: shouldn't break line here

File: do_math_exercises/merge_tools.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
+#Hackeerrank: https://www.hackerrank.com/challenges/merge-the-tools/problem?isFullScreen=true
+
+def merge_the_tools(string, k):
+    def sub_string_to_string(sub_string):
+        sub_string = list(sub_string)
+        set_sub_string = []
```
Comment (thengoc-bui-goldenowl) [inline do_math_exercises/merge_tools.py:6]: do not initial

**Review by thengoc-bui-goldenowl — APPROVED** [review]

**Pull request merged**